docs/hugo/content/design/ADR-2023-02-Adoption-Policy.md

```diff
@@ -0,0 +1,258 @@
+| detach-on-delete   | skip-if-exists             | No                                | Manages the resource, detaches on delete |
+| skip               | skip-if-exists             | Yes                               | Skips managing the resource              |
+| skip               | skip-if-exists             | No                                | Skips managing the resource              |
+| Unspecified        | detach-on-delete-if-exists | Yes                               | Manages the resource, detaches on delete |
+| Unspecified        | detach-on-delete-if-exists | No                                | Fully manages the resource               |
+| detach-on-delete   | detach-on-delete-if-exists | Yes                               | Manages the resource, detaches on delete |
+| detach-on-delete   | detach-on-delete-if-exists | No                                | Manages the resource, detaches on delete |
+| skip               | detach-on-delete-if-exists | Yes                               | Skips managing the resource              |
+| skip               | detach-on-delete-if-exists | No                                | Skips managing the resource              |
+
+¹: Actual behavior depends on earlier sections of this design
+
+**Advantages:**
+- ???
+
+**Disadvantages:**
+- Users need to understand this complicated table to figure out what is actually happening to their resource, it's not
+  immediately obvious based on the annotations.
+- There's no way to tell which option was actually chosen for users.
+- We need to somehow remember which option we chose so we don't think that a resource we created needs to be adopted after
+  a crash or pod restart. That further increases the number of annotations used by this feature...
+
+##### Option 4.2: reconcile-policy-if-exists annotation
+
+A new property `RECONCILE_POLICY_IF_EXISTS` in the operator global secret, and an annotation on each
+resource `serviceoperator.azure.com/reconcile-policy-if-exists` would allow users to control this behavior.
+
+Possible `reconcile-policy-if-exists`'s are the same as the existing `reconcile-policy`
+- `manage`: Corresponding to scenario 1 above.
+- `detach-on-delete`: Corresponding to scenario 2 above.
+- `skip`: Corresponding to scenario 3 above.
+
+The implementation of `reconcile-policy-if-exists` would basically be a delayed defaulting of `reconcile-policy` applied
+when reconciliation ran for the first time. The operator would check if the resource already exists and write the corresponding 
+`reconcile-policy` annotation to the resource if the resource existed.
+
+**Advantages:**
+- Uses a concept we already have.
+
+**Disadvantages:**
+- Possibility of collision between user specified `reconcile-policy` and the `reconcile-policy` written by the operator because
+  of `reconcile-policy-if-exists`. 
+  - We can mitigate this issue by rejecting some configurations with a webhook, although we have to be careful how we craft
```

Comment:
Would something like `reconcile-policy: skip`, `reconcile-policy-if-not-exists: manage` be possible to express using this method? What would happen in that case for a resource that does not already exist? 

(Asking generally, I don't think we'd do something like that in CAPZ)

Reply:
Great question. I've expanded the disadvantages section to talk about two types of possible mitigation here (up from 1).

The second option is to allow both and have `reconcile-policy-if-exists` always overwrite `reconcile-policy`. The only downside I see to this is that it could cause some gitops flows issue as we might overwrite the annotation they provided with the annotation in `reconcile-policy-if-exists`, causing their checked-in spec to differ from the active spec (annotation mismatch). It's a minor issue but at the same time I am not clear on what exact scenarios users would really want to set both in practice, for example your example scenario feels pretty contrived. One that seems possibly valid is: `reconcile-policy: detach-on-delete` and  `reconcile-policy-if-exists: skip`...

I've updated the decision section to lean in the direction of the first option (webhook prevention) as we can always expand it later but we can't undo letting users set both if we do it now, but I'm open to other opinions on that.